### ISSUE TITLE
Long rounding issue

### ISSUE THREAD

**Issue by ostrovskyde** [hawtio/hawtio-integration#108]
In case JMX operation has java.lang.Long or primitive long argument, Hawtio is rounding large numbers.
E.g. 282784413042312285 is being transformed to 282784413042312300, so the actual execution takes latter.

Please use long.js to deal with Long type arguments.

**Comment by ostrovskyde** [comment]
@tadayosi kindly share your thoughts on this

**Comment by tadayosi** [comment]
@ostrovskyde Thanks for the report!  I agree that Hawtio should handle large numbers as-is and not round them. That said, at this moment it's not clear to me whether it's an issue in Hawtio or Jolokia. Can you test it with the Jolokia endpoint using `curl`?  If it still reproduces the issue, then we should rather report it to https://github.com/rhuss/jolokia/issues.

**Comment by ostrovskyde** [comment]
@tadayosi in DevTools I saw that the actual value being send in the request body was already rounded. So the issue is definitely on the UI side.

**Comment by tadayosi** [comment]
Thanks. Let's see if we can fix it easily.

**Comment by yuhusolutions** [comment]
Hi everyone, I looked into this and I confirm the issue stems from the Jolokia JS lib. 
Jolokia delegates the Ajax call to JQuery, using the native JSon parser. The built in Json  parser box the numeric value into a Number, which max value is shorter than Long (double-precision floats).
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_VALUE
This is not an uncommon issue in Javascript when dealing with large numbers. 

Suggestion would be to provide a custom JSON parser and parse the numeric value into BigInt
For instance: https://www.npmjs.com/package/json-bigint

```js
console.log(Number.parseFloat("9223372036854775807"))
9223372036854776000
```

![Screenshot from 2022-02-05 19-37-16](https://user-images.githubusercontent.com/29959338/152666849-9ce5579b-4b38-4815-8d06-936b39ab4d0e.png)

**Comment by tadayosi** [comment]
Fixed at latest Hawtio 4.0.0
